mib-generator/cw_mib_generator.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/python
+
+# Copyright (C) Metaswitch Networks 2017
+# If license terms are provided to you in a COPYING file in the root directory
+# of the source code repository by which you are accessing this code, then
+# the license outlined in that COPYING file applies to your use.
+# Otherwise no rights are granted except for those provided to you by
+# Metaswitch Networks in a separate written agreement.
+
+"""Metaswitch Clearwater MIB Generator
+Usage:
+  cw_mib_generator.py [--cwc-mib-dir=DIR]
+
+Options:
+  --cwc-mib-dir=DIR  Directory containing Clearwater Core MIB fragment
+"""
+
+import re
+import sys
+import os.path
+from string import Template
+from docopt import docopt
+
+# Project Clearwater MIB paths
+COMMON_MIB_PATH = "./CLEARWATER-MIB-COMMON"
+PC_EXTRAS_PATH = "./CLEARWATER-MIB-PC-EXTRAS"
+PC_MIB_OUTPUT_PATH = "../PROJECT-CLEARWATER-MIB"
+
+# Clearwater Core MIB names
+CWC_EXTRAS_NAME = "CLEARWATER-MIB-CWC-EXTRAS"
+CWC_MIB_OUTPUT_NAME = "METASWITCH-CLEARWATER-CORE-MIB"
+
+# Statement added to top of auto-generated MIBs
+EDIT_STATEMENT = "THIS MIB IS BUILT FROM A TEMPLATE - DO NOT EDIT DIRECTLY!"
+
+
+def print_err_and_exit(error_text):
+    # Prints an error message with the specified text, and exits with code 1.
+    sys.stderr.write("ERROR: {}\nFailed to generate MIB, exiting\n"
+                     .format(error_text))
+    sys.exit(1)
```

Comment:
Why not pass the error message to `sys.exit` instead of printing it out separately?